server/models/Profile.js

```diff
@@ -64,6 +64,10 @@ const profileSchema = new Schema({
   requests: [{
     type: Schema.Types.ObjectId,
     ref: "Request"
+  }],
+  notifications: [{
```

Comment:
With the way you have to go through notifications, perhaps this schema might help  https://mongoosejs.com/docs/schematypes.html#maps

The string could be title \n message? Just a thought